### PR TITLE
Add similarity feature to avoid duplicate object creation

### LINKED ISSUE #106: Similarity feature
Add a similarity feature that attempts to minimize duplicates by suggesting exisiting data objects when new ones are created. Applicable to:
- [x] People creation - search for people with a similar name
- [x] Locations creation - search for locations with a similar name
- [x] Positions creation - search for positions with a similar name

Initially we will do a simple text search when creating one of these objects and show matches as "Possible duplicates".

Later refinement might include:
- [ ] Better fuzzy matching by using similarity queries supported by the underlying database
- [ ] Match locations within a close geo distance

### PR DESCRIPTION
**Pull request by cemalettin-work**
Add the option to show possible duplicates when creating new people, positions or locations,

Closes #106

#### User changes
- None

#### Super User changes
- When creating new people, positions or locations, possible duplicates can be shown to prevent creating objects that already exist in the database.

#### Admin changes
- When creating new people, positions or locations, possible duplicates can be shown to prevent creating objects that already exist in the database.

#### System admin changes
- [ ] anet.yml or anet-dictionary.yml needs change
- [ ] db needs migration
- [ ] documentation has changed
- [ ] graphql schema has changed

### Checklist
  - [x] Described the user behavior in PR body
  - [x] Referenced/updated all related issues
  - [x] commits follow a `repo#issue: Title` title format and [these 7 rules](https://chris.beams.io/posts/git-commit/)
  - [x] commits have a [clean history](https://epage.github.io/dev/commits/), otherwise PR may be squash-merged
  - [ ] Added and/or updated unit tests
  - [x] Added and/or updated e2e tests
  - [ ] Added and/or updated data migrations
  - [ ] Updated documentation
  - [x] Resolved all build errors and warnings
  - [ ] Opened debt issues for anything not resolved here

### CODE REVIEW

File: client/src/pages/people/Form.js [hunk 1]
```diff
@@ -216,6 +220,19 @@ const PersonForm = ({ edit, title, saveText, initialValues }) => {
                       />
                     </Col>
                   </Col>
+                  {values.firstName.length >= 2 && values.lastName.length >= 2 && (
```
Comment (gjvoosten) [inline client/src/pages/people/Form.js:223]: Maybe define a constant instead of hardcoding `2` here?

File: client/src/pages/locations/Form.js [hunk 2]
```diff
@@ -149,6 +153,21 @@ const LocationForm = ({ edit, title, initialValues }) => {
                   name="name"
                   component={FieldHelper.InputField}
                   disabled={!canEditName}
+                  extraColElem={
+                    values.name.length >= 3 ? (
```
Comment (gjvoosten) [inline client/src/pages/locations/Form.js:157]: Maybe define a constant instead of hardcoding `3` here?

File: client/src/pages/positions/Form.js [hunk 3]
```diff
@@ -257,11 +261,26 @@ const PositionForm = ({ edit, title, initialValues }) => {
...
                 />
 
-                <FastField
+                <Field
                   name="name"
                   component={FieldHelper.InputField}
                   label={Settings.fields.position.name}
                   placeholder="Name/Description of Position"
+                  extraColElem={
+                    values.name.length >= 3 ? (
```
Comment (gjvoosten) [inline client/src/pages/positions/Form.js:270]: Maybe define a constant instead of hardcoding `3` here?

**Review by gjvoosten — CHANGES_REQUESTED** [review]

**Review by gjvoosten — CHANGES_REQUESTED** [review]

File: client/src/pages/locations/Form.js [hunk 4]
```diff
@@ -149,6 +154,21 @@ const LocationForm = ({ edit, title, initialValues }) => {
                   name="name"
                   component={FieldHelper.InputField}
                   disabled={!canEditName}
+                  extraColElem={
+                    !edit && values.name.length >= MIN_CHARS_FOR_DUPLICATES ? (
```
[the remark this answers is not in the document]
Comment (anilsonmez-simsoft) [reply client/src/pages/locations/Form.js:158]: Then I get a warning. `extraColElem` doesn't like the boolean input and expects an object.

[the remark this answers is not in the document]
Comment (gjvoosten) [reply client/src/pages/locations/Form.js:158]: Ah yes, I did too. :nerd_face:
No change needed then.

**Review by gjvoosten — APPROVED** [review]

**Pull request merged**